### ISSUE TITLE
[docs] rm class inheritance diagram from UG

### ISSUE THREAD

**Issue by arporter** [stfc/PSyclone#771]
Section 7.2 of the User Guide currently contains a class-inheritance diagram for the Dynamo API. This information is now automatically generated as part of the reference guide and does not belong in a user guide anyway. Therefore we can:

1. Remove this part of the user guide.
2. Remove the pyreverse-specific parts of the source code (to be agreed with @rupertford, @sergisisi and @hiker)?

**Comment by rupertford** [comment]
We have used pyreverse to generate a hierarchy suitable for viewing and presenting a simplified subset of the classes in the past. I admit we don't use that slide anymore so perhaps it is not useful. Would it be useful for all these publications we are going to write? :-).
Oh, and it is @sergisiso :-).

**Comment by sergisiso** [comment]
I have to say I find this diagrams useful to understand the APIs even if it's a simplified or conceptual version, although I understand it is tricky to maintain them when there are changes. By automatically generated do you mean the Doxygen ones or something else?

**Comment by arporter** [comment]
Yes, I meant the Doxygen ones although I admit that they don't have as much information as those from pyreverse. We can leave the support for pyreverse and just refactor the documentation if people prefer? In fact, we could try and get the pyreverse-generated pics incorporated into the RTD build process.

**Comment by hiker** [comment]
I would be in favour of automatically creating the graph as well (and otherwise leave a copy there - it might become something like the documentation pdf - whenever we merge a branch, we have to re-create the pdf, then we would also need to re-create the image(s) for the inheritance diagrams.

**Comment by sergisiso** [comment]
There is at least one [sphinx-pyreverse](https://pypi.org/project/sphinx-pyreverse/) plugin, so it may be possible to automate it. If it works I also prefer it as the Doxygen does not show the associativity connections.
